lib/ansible/modules/cloud/vmware/vmware_guest_disk.py

```diff
@@ -0,0 +1,604 @@
+                current_disk['scsi_controller'] = disk['scsi_controller']
+            elif 'scsi_controller' not in disk:
+                self.module.fail_json(msg="Please specify 'scsi_controller' under disk parameter"
+                                          " at index [%s], which is required while creating disk." % disk_index)
+            # Check for disk unit number
+            if 'unit_number' in disk:
+                if disk['unit_number'] not in range(0, 16):
+                    self.module.fail_json(msg="Invalid Disk unit number ID specified for disk [%s] at index [%s],"
+                                              " please specify value between 0 to 15"
+                                              " only (excluding 7)." % (disk['unit_number'], disk_index))
+
+                if disk['unit_number'] == 7:
+                    self.module.fail_json(msg="Invalid Disk unit number ID specified for disk at index [%s],"
+                                              " please specify value other than 7 as it is reserved"
+                                              "for SCSI Controller" % disk_index)
+                current_disk['disk_unit_number'] = disk['unit_number']
+
+            elif 'unit_number' not in disk:
+                self.module.fail_json(msg="Please specify 'unit_number' under disk parameter"
+                                          " at index [%s], which is required while creating disk." % disk_index)
+
+            # Type of Disk
+            disk_type = disk.get('type', 'thick').lower()
+            if disk_type not in ['thin', 'thick', 'eagerzeroedthick']:
+                self.module.fail_json(msg="Invalid 'disk_type' specified for disk index [%s]. Please specify"
+                                          " 'disk_type' value from ['thin', 'thick', 'eagerzeroedthick']." % disk_index)
+            current_disk['disk_type'] = disk_type
+
+            # SCSI Controller Type
+            scsi_contrl_type = disk.get('scsi_type', 'paravirtual').lower()
+            if scsi_contrl_type not in self.scsi_device_type.keys():
+                self.module.fail_json(msg="Invalid 'scsi_type' specified for disk index [%s]. Please specify"
+                                          " 'scsi_type' value from ['%s']" % (disk_index,
+                                                                              "', '".join(self.scsi_device_type.keys())))
+            current_disk['scsi_type'] = scsi_contrl_type
+
+            disks_data.append(current_disk)
+        return disks_data
+
+    def get_recommended_datastore(self, datastore_cluster_obj=None):
+        """
+        Function to return Storage DRS recommended datastore from datastore cluster
+        Args:
+            datastore_cluster_obj: datastore cluster managed object
+
+        Returns: Name of recommended datastore from the given datastore cluster
+
+        """
+        if datastore_cluster_obj is None:
+            return None
```

Comment:
Are you should you want to handle the case where datastore_cluster_obj is None ; I find this misleading.
Also get_recommended_datastore is only called once in your code, and datastore_cluster_obj can't be None.

Also, please document in the docstring that get_recommended_datastore return None if no datastore is found.

Reply:
OK.

Reply:
`def get_recommended_datastore(self, datastore_cluster_obj=None):`
→
`def get_recommended_datastore(self, datastore_cluster_obj):`